### PR TITLE
feat: (TNLT-5967) Add tooltip for bylines with authors and refactor tooltip component

### PR DESCRIPTION
**Pull request by aronshamash**
- https://nidigitalsolutions.jira.com/browse/TNLT-5967
- Add tooltip for bylines with authors and refactors tooltip component
- Tooltip now takes more props to make it more flexible. 
- The author tooltip is not shown on the in-depth article template due to layout issues relating to zIndex in React Native.
- The commenting tooltip has now been placed to the right of the button.

### iPad and iPhone 8  Comment article template
Screenshot 2020-11-24 at 14 01 17

 
### iPad Mini Main standard article template
Screenshot 2020-11-25 at 14 59 03

### iPad and iPhone 8  Magazine standard article template
Screenshot 2020-11-24 at 14 34 40

### iPad  Indepth article template
Screenshot 2020-11-24 at 14 35 36

### Android  Pixel C
Screenshot 2020-11-25 at 12 54 42

### CODE REVIEW

File: packages/article-byline/src/article-byline-with-links.js [hunk 1]
```diff
@@ -32,12 +34,49 @@ const AuthorComponent = ({
...
+    tooltipOffsetY,
+    tooltips,
+  } = props;
+
   const textStyle = centered ? [styles.text, styles.centered] : styles.text;
-  return renderByline(withTrackEvents(AuthorComponent), ast, textStyle, props);
-};
+  const tooltipType = "profile";
+  const showTooltip =
+    !disableTooltip && hasAuthorData(ast) && tooltips.includes(tooltipType);
```
Comment (c-j-j) [inline packages/article-byline/src/article-byline-with-links.js:50]: are we showing tooltips on mobile? if not, I assume disableTootip is set to false?

File: packages/tooltip/tooltip.tsx [hunk 2]
```diff
@@ -61,30 +73,16 @@ const Tooltip: React.FC<Props> = ({
...
-              <Text style={[styles.text, leftAligned && styles.textLeft]}>
-                {content}
-              </Text>
-              <View
-                style={[
-                  styles.arrow,
-                  { left: calculateArrowPosition(alignment, width) },
-                ]}
-              />
+              <Text style={styles.text}>{content}</Text>
```
Comment (c-j-j) [inline packages/tooltip/tooltip.tsx:78]: Maybe want to set `allowFontScaling` to false here? I assume we wouldn't want giant text in this tooltip if the user has it set on their device?

File: packages/article-magazine-standard/src/article-magazine-standard.js [hunk 3]
```diff
@@ -54,24 +62,22 @@ class ArticleMagazineStandard extends Component {
...
               getImageCrop={getStandardTemplateCrop}
               onImagePress={onImagePress}
               onVideoPress={onVideoPress}
               renderCaption={({ caption }) => <CentredCaption {...caption} />}
-              style={[
-                styles.leadAssetContainer,
-                isTablet && styles.leadAssetContainerTablet,
-                isTablet && styles.tabletContainer,
-              ]}
+              style={[{ zIndex: 0 }]}
```
Comment (c-j-j) [inline packages/article-magazine-standard/src/article-magazine-standard.js:77]: can we object instead of array

**Review by c-j-j — APPROVED** [review]
Looks great - nice work

File: packages/article-magazine-standard/src/article-magazine-standard.js (hunk 3 above)
Comment (aronshamash) [reply packages/article-magazine-standard/src/article-magazine-standard.js:77]: good catch this some debug code left in. will revert

File: packages/tooltip/tooltip.tsx (hunk 2 above)
Comment (aronshamash) [reply packages/tooltip/tooltip.tsx:78]: good point. yes will add

File: packages/article-byline/src/article-byline-with-links.js (hunk 1 above)
Comment (aronshamash) [reply packages/article-byline/src/article-byline-with-links.js:50]: `disableTooltip` is prop passed for any templates or component that need to disable it. Within the tooltip component there is a check to only show the tooltip part on tablets.

**Pull request merged**